phys/module_mp_jensen_ishmael.F

```diff
@@ -0,0 +1,4701 @@
+  ! growth, sublimation, riming, and melting. Aggregates are parameterized as a third ice species        !
+  ! because the shape and density change that occurs during aggregation is not necessarily smooth.       !
+  ! Predicting two ice particle dimensions requires two prognostic variables. Thus mass, number, and two !
+  ! size-related mixing ratio are predicted for each ice species.  That makes 12 prognostic variables    !
+  ! for ice. Just throw more processors at it. Rain mass and number and cloud mass are also predicted    !
+  ! See Jensen et al. 2017 (JAS), Jensen et al. 2018 (MWR), and Jensen et al. 2018 (JAS) for details.    !
+  !------------------------------------------------------------------------------------------------------!
+
+  use module_wrf_error   
+
+  implicit none ! You are welcome
+
+  !.. Constants
+  real, private, parameter ::     &
+       PI          = 3.14159265,  &  ! Pi = 3.14159265... until forever
+       G_HOME      = 9.8,         &  ! Gravity (m s^-1)
+       RD          = 287.15,      &  ! Dry air gas constant (J kg^-1 K^-1)
+       RV          = 461.5,       &  ! Moist air gas constant (J kg^-1 K^-1)
+       CP          = 1004.0,      &  ! Heat capacity Air (J kg^-1 K^-1)
+       CPW         = 4218.0,      &  ! Heat capacity Water (J kg^-1 K^-1)
+       RCP         = 0.285856574, &  ! RD / CP
+       RHOW        = 1000.0,      &  ! Water density (kg m^-3)
+       T0          = 273.15,      &  ! STP temperature (K)
+       P0          = 101325.0,    &  ! STP pressure (Pa)
+       RHOI        = 920.0,       &  ! Bulk ice density (kg m^-3)
+       R0          = 1.27494         ! Air density at 1000 mb, T=T0 (kg m^-3)
+
+  !.. The lookup table arrays are itab for ice-cloud water collection 
+  !.. and itabr for ice-rain collection. The lookup tables are read in 
+  !.. jensen_ishmael_init which occurs once.
+  real, private, allocatable, dimension(:,:,:,:,:) :: itab, itabr
+
+  !.. Inherent growth ratio data is stored in the igrdata array
+  !.. The gamma_tab is a tabulation of the gamma function needed
+  !.. because ice particle shape evolves. 
+  real, private, allocatable, dimension(:) :: igrdata, gamma_tab
+
+contains
+  !------------------------------------------------------------------------------------------------------!
+  !------------------------------------------------------------------------------------------------------!
+
+  subroutine jensen_ishmael_init
+    implicit none
+    
+    integer :: i1, i2, i3, i4, count
+  !.. Be careful when using a moving nest or any instance
+  !.. where micro_init is called frequently as this may
+  !.. slow down the code. 
+
+  !.. Read in ice-cloud water collection lookup table
+    if(.not.allocated(itab)) then 
+       allocate(itab(51,51,51,11,2))
+
+       open(20, FILE='ishmael-qi-qc.dat')
+       do i1 = 1,51
+          do i2 = 1,51
+             do i3 = 1,51
+                do i4 = 1,11
+                   read(20, *) itab(i1,i2,i3,i4,1), &
```

Comment:
The read statements here imply that the look-up tables are stored as formatted text, right? On the one hand, this avoids any endian issues when reading on various architectures, but it does mean that the files will likely be several times larger than they would be if the data were stored in binary.

Reply:
@mgduda I can convert to binary. Is there an easy way to ensure proper endian?

Reply:
I think WRF expects all unformatted files to be in big-endian format. In this case, when writing the unformatted look-up tables, you can compile your code with a byteswap flag that is compiler-dependent:
* ifort: `-convert big_endian`
* gfortran: `-fconvert=big-endian`
* pgf90: `-byteswapio`